input/otlp/logs.go

```diff
@@ -239,3 +245,26 @@ func (c *Consumer) convertLogRecord(
 
 	return event
 }
+
+// Sanitize the datastream fields (dataset, namespace) to apply restrictions
+// as outlined in https://www.elastic.co/guide/en/ecs/current/ecs-data_stream.html
+func sanitizeDataStreamField(field string) string {
+	field = strings.ToLower(field)
+	field = strings.Map(replaceReservedRune, field)
+	// Cannot start with  _, +
+	// https://github.com/elastic/ecs/blob/main/rfcs/text/0009-data_stream-fields.md
```

Comment:
I'd prefer a link to the actual ecs definition rather than rfc. The rfc is outdated and says `type` should either be `logs` or `metrics`.